test/parallel/test-quic-openstream-pending.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+const common = require('../common');
+if (!common.hasQuic)
+  common.skip('missing quic');
+
+// Test that opening a stream works even if the session isn’t ready yet.
+
+const assert = require('assert');
+const quic = require('quic');
+
+const fixtures = require('../common/fixtures');
+const key = fixtures.readKey('agent1-key.pem', 'binary');
+const cert = fixtures.readKey('agent1-cert.pem', 'binary');
+const ca = fixtures.readKey('ca1-cert.pem', 'binary');
+
+const server = quic.createSocket({ port: 0, validateAddress: true });
+
+server.listen({
+  key,
+  cert,
+  ca,
+  rejectUnauthorized: false,
+  maxCryptoBuffer: 4096,
+  alpn: 'meow',
+  maxStreamsUni: 100
+});
+
+server.on('session', common.mustCall((session) => {
+  session.on('stream', common.mustCall((stream) => {
+    let data = '';
+    stream.setEncoding('utf8');
+    stream.on('data', (chunk) => data += chunk);
+    stream.on('end', common.mustCall(() => {
+      assert.strictEqual(data, 'Hello!');
+      session.close();
+      server.close();
+    }));
+  }));
+
+  session.on('close', common.mustCall());
+}));
+
+server.on('ready', common.mustCall(() => {
+  const client = quic.createSocket({
+    port: 0,
+    client: {
+      key,
+      cert,
+      ca,
+      alpn: 'meow'
+    }
+  });
+
+  const req = client.connect({
+    address: 'localhost',
+    port: server.address.port
+  });
+
+  const stream = req.openStream({ halfOpen: true });
+  stream.end('Hello!');
+
+  assert.strictEqual(stream.pending, true);
+  stream.on('ready', common.mustCall(() => {
```

Comment:
In the `ready` event, I'm wondering if the stream ID shouldn't be passed off to the callback, just as a helpful bit of context.

Reply:
I’m not sure if that really provides more context – it is available as `stream.id` (or `this.id`) in the event handler – but I have added a note about it in the documentation.

Reply:
Resolved in https://github.com/nodejs/quic/pull/187/commits/340eaceaebf374dcdd2cf41f920974c30cdb85d1